vowpalwabbit/example.h

```diff
@@ -49,11 +49,58 @@ typedef union
   float prob;  // for --probabilities --csoaa_ldf=mc
 } polyprediction;
 
+class gd_prediction_cache
+{
+  std::vector<polyprediction> _cache;
+
+ public:
+  bool inline get_value(uint64_t ft_offset, polyprediction& pred)
+  {
+    if (_cache.size() > ft_offset)
+    {
+      pred = _cache[ft_offset];
+      return true;
+    }
+    return false;
+  }
+
+  void inline set_value(uint64_t ft_offset, const polyprediction& pred)
+  {
+    if (_cache.size() <= ft_offset)
+    {
+      _cache.resize(ft_offset + 1);
+    }
+    _cache[ft_offset] = pred;
```

Comment:
Copying a union seems like a dangerous pattern to be using here

Reply:
This is was an experiment.  Not used.  Removed.